senior_project_dea-main/server/server.js

```diff
@@ -102,16 +102,16 @@ server.put("/user/update/:id", async(req,res) => {
         //Set _id to the value given in url under :id
         const _id = req.params.id;
 
-        if (req.body.password != undefined) {
+        if (req.body.password !== undefined) {
             req.body.password = await bcrypt.hash(req.body.password, 10);
         }
 
-        if (req.body.email != undefined) {
+        if (req.body.email !== undefined) {
             const email = req.body.email
             const existingUser = await User.findOne({email});
 
             //If another user besides the one we're updating has the same email
-            if (existingUser && existingUser._id != _id) {
+            if (existingUser && existingUser._id !== _id) {
```

Comment:
did you test this? I seem to remember it not working for me with strict inequality, but it could have been something else going wrong..

Reply:
Worked for me. Granted, was not my main focus of testing for this update. Feel free to change it back if it breaks.